### DISCUSSION TITLE
Last Good Check DB on master database didn't get update

### DISCUSSION THREAD

**Discussion by chucklor** [trimble-oss/dba-dash#914] [Q&A] [answered]
I have the master, tempdb, model, and msdb all in one backup agent job. However, the status of the master database doesn't seem to get updated or correct even though all of these databases backup successfully. What am I missing? Thanks!
![image](https://github.com/trimble-oss/dba-dash/assets/20845188/bed445bd-0a1e-403d-bd66-dafe6e2d299a)

**Comment by DavidWiseman** [comment, marked as answer]
The Last Good Check DB gets updated when you run DBCC CHECKDB.  It seems like this isn't been updated.  You can run this on your monitored instance to check:

```
SELECT database_id,name,CAST(DATABASEPROPERTYEX(name,'LastGoodCheckDbTime') as DATETIME) as LastGoodCheckDbTime
FROM sys.databases
```

Note: The collection uses [this query](https://raw.githubusercontent.com/trimble-oss/dba-dash/main/DBADash/SQL/SQLLastGoodCheckDB.sql) which accounts for older versions of SQL, but if you are on a reasonably modern version of SQL it's the query above that will run.

To fix the problem you can run this on your monitored instance:
```
USE master
GO
DBCC CHECKDB WITH NO_INFOMSGS,ALL_ERRORMSGS
```
You would need to investigate why it's not been run from your agent job though.

Once the DBCC check has run on your monitored instance, the last good check DB collection runs every 1hr by default.  You will need to wait for the next collection or trigger it to run manually (new feature in [3.5](https://dbadash.com/blog/whats-new-in-3.5/) or restart the service)

Hope this helps
